mmv1/third_party/terraform/services/compute/resource_compute_router_bgp_peer_test.go

```diff
@@ -494,6 +552,268 @@ resource "google_compute_router_peer" "foobar" {
 `, routerName, routerName, routerName, routerName, routerName, routerName, routerName, routerName, routerName)
 }
 
+func testAccComputeRouterPeerWithMd5AuthKey(routerName string) string {
+	return fmt.Sprintf(`
+  resource "google_compute_network" "foobar" {
+    name = "%s-net"
+    auto_create_subnetworks = false
+  }
+  
+  resource "google_compute_subnetwork" "foobar" {
+    name          = "%s-subnet"
+    network       = google_compute_network.foobar.self_link
+    ip_cidr_range = "10.0.0.0/16"
+    region        = "us-central1"
+  }
+  
+  resource "google_compute_address" "foobar" {
+    name   = "%s"
+    region = google_compute_subnetwork.foobar.region
+  }
+  
+  resource "google_compute_ha_vpn_gateway" "foobar" {
+    name    = "%s-gateway"
+    network = google_compute_network.foobar.self_link
+    region  = google_compute_subnetwork.foobar.region
+  }
+  
+  resource "google_compute_ha_vpn_gateway" "foobar1" {
+    name    = "%s-gateway1"
+    network = google_compute_network.foobar.self_link
+    region  = google_compute_subnetwork.foobar.region
+  }
+  
+  resource "google_compute_external_vpn_gateway" "external_gateway" {
+    name            = "%s-external-gateway"
+    redundancy_type = "SINGLE_IP_INTERNALLY_REDUNDANT"
+    description     = "An externally managed VPN gateway"
+    interface {
+      id         = 0
+      ip_address = "8.8.8.8"
+    }
+  }
+  
+  resource "google_compute_router" "foobar" {
+    name    = "%s"
+    region  = google_compute_subnetwork.foobar.region
+    network = google_compute_network.foobar.self_link
+    bgp {
+      asn = 64514
+    }
+  }
+  
+  resource "google_compute_vpn_tunnel" "foobar" {
+    name               = "%s"
+    region             = google_compute_subnetwork.foobar.region
+    vpn_gateway = google_compute_ha_vpn_gateway.foobar.id
+    peer_external_gateway           = google_compute_external_vpn_gateway.external_gateway.id
+    peer_external_gateway_interface = 0  
+    shared_secret      = "unguessable"
+    router             = google_compute_router.foobar.name
+    vpn_gateway_interface           = 0
+  }
+  
+  resource "google_compute_vpn_tunnel" "foobar1" {
+    name               = "%s1"
+    region             = google_compute_subnetwork.foobar.region
+    vpn_gateway = google_compute_ha_vpn_gateway.foobar1.id
+    peer_external_gateway           = google_compute_external_vpn_gateway.external_gateway.id
+    peer_external_gateway_interface = 0  
+    shared_secret      = "unguessable"
+    router             = google_compute_router.foobar.name
+    vpn_gateway_interface           = 0
+  }
+  
+  resource "google_compute_router_interface" "foobar" {
+    name       = "%s"
+    router     = google_compute_router.foobar.name
+    region     = google_compute_router.foobar.region
+    vpn_tunnel = google_compute_vpn_tunnel.foobar.name
+  }
+  
+  resource "google_compute_router_interface" "foobar1" {
```

Comment:
To fix the non-determinism can you add depends_on blocks to all of these fine-grained resources that depend on a single parent? So each router_interface that uses a single router should depend on the previous router_interface to give us a deterministic chain of execution during tests. Same with router peers